src/main/java/kernbeisser/Windows/SoloShoppingMask/SoloShoppingMaskView.java

```diff
@@ -44,4 +49,10 @@ public void initialize(SoloShoppingMaskController controller) {
     public IconCode getTabIcon() {
         return FontAwesome.SHOPPING_CART;
     }
+
+    @Override
+    public boolean processKeyboardInput(KeyEvent e) {
+        return shoppingMaskUIView.processKeyboardInput(e);
```

Comment:
Is this not a controller function? But I think I am not here the person which is able to decide this ;)

Reply:
I thought about that. I think, the barcode collection  is quite similar to a control and therefore ui. The code basically reacts on keyboard events and stores the typed keys in a variable.  Only the code, that takes the barcode and decides what to do with it, is controller -> controller.processBarcode(barcode) I.e. the consumer of the barcode